src/ocnews_api.cpp

```diff
@@ -21,56 +18,34 @@ typedef std::unique_ptr<json_object, decltype(*json_object_put)> json_uptr;
 typedef std::unique_ptr<CURL, decltype(*curl_easy_cleanup)> curl_uptr;
 
 ocnews_api::ocnews_api(configcontainer* c) : remote_api(c) {
-	std::string user = cfg->get_configvalue("ocnews-login");
-	bool flushed = false;
-
-	if (user == "") {
-		std::cout << std::endl;
-		std::cout.flush();
-		flushed = true;
-		std::cout << "Username for ocNews: ";
-		std::cin >> user;
-
-		if (user.empty())
-			LOG(level::CRITICAL, "ocnews_api::ocnews_api: No user set");
-	}
-
-	std::string pass = cfg->get_configvalue("ocnews-password");
-	std::string pass_file = cfg->get_configvalue("ocnews-passwordfile");
-	std::string pass_eval = cfg->get_configvalue("ocnews-passwordeval");
-	if ( pass == "" ) {
-		if ( pass_file != "" ) {
-			pass = read_password(pass_file);
-		} else if ( pass_eval != "") {
-			pass = eval_password(pass_eval);
-		} else {
-			if (!flushed) {
-				std::cout << std::endl;
-				std::cout.flush();
-			}
-			// Find a way to do this in C++ by removing cin echoing.
-			pass = std::string( getpass("Password for ocNews: ") );
-		}
-
-		if (pass.empty())
-			LOG(level::CRITICAL, "ocnews_api::ocnews_api: No password set");
-	}
-
-	auth = user+":"+pass;
 	server = cfg->get_configvalue("ocnews-url");
 	verifyhost = cfg->get_configvalue_as_bool("ocnews-verifyhost");
-
-	if (server.empty())
```

Comment:
Similarly to `cookie-cache` comment in Newsblur API: split this into a separate issue, please.